compiler/damlc/lib/DA/Cli/Damlc/Packaging.hs

```diff
@@ -432,33 +509,19 @@ mbErr err = maybe (hPutStrLn stderr err >> exitFailure) pure
 lfVersionString :: LF.Version -> String
 lfVersionString = DA.Pretty.renderPretty
 
-
--- | The graph will have an edge from package A to package B if A depends on B.
+-- | The graph will have an edge from package A to package B:
+--   - if A depends on B _OR_
+--   - if A is a data-dependency, B is a (regular) dependency, and B doesn't depend on any data-dependencies.
 buildLfPackageGraph
```

Comment:
This is the core change in this PR. 

Previously, the graph only had nodes for the _data-dependencies_ (which included stable packages) of the project being built, and edges from a package to its dependencies. 

Now, it has nodes for all dependencies of the project (including data-dependencies, plain dependencies, builtin packages (currently `daml-stdlib` and `daml-prim`) and stable packages), and edges from a package to its dependencies _and_ from (non stable package) data-dependencies to the subset of (plain) dependencies which do not in turn depend on any data-dependency.

Reply:
As a consequence, some of the `data-dependency`-specific logic that happened here had to move to `installDataDep`, in particular the call to `DA.Daml.Compiler.DataDependencies.generateSrcPkgFromLf`

Reply:
It would be great if this comment would be extended for the motivation why we want this exact graph. Something along the lines @sofiafaro-da is giving above for the type class rewriting in data-dependencies.

Reply:
> from (non stable package) data-dependencies to the subset of (plain) dependencies which do not in turn depend on any data-dependency.
I assume these (plain) dependencies are not allowed to _transitively_ depend any data-dependency. It seems that's what the code does, but I think it would be good to make this explicit.